rmw_fastrtps_shared_cpp/src/rmw_node.cpp

```diff
@@ -42,45 +47,60 @@ __rmw_create_node(
   const char * name,
   const char * namespace_)
 {
-  if (!name) {
-    RMW_SET_ERROR_MSG("name is null");
+  assert(nullptr != context);
+  assert(nullptr != context->impl);
+  assert(identifier == context->implementation_identifier);
+  static_cast<void>(identifier);
+
+  int validation_result = -1;
+  rmw_ret_t ret = rmw_validate_node_name(name, &validation_result, nullptr);
+  if (RMW_RET_OK != ret || RMW_NODE_NAME_VALID != validation_result) {
+    const char * reason = RMW_RET_OK == ret ?
+      rmw_node_name_validation_result_string(validation_result) :
+      rmw_get_error_string().str;
+    RMW_SET_ERROR_MSG_WITH_FORMAT_STRING("invalid node name: %s", reason);
     return nullptr;
   }
-
-  if (!namespace_) {
-    RMW_SET_ERROR_MSG("namespace_ is null");
+  validation_result = -1;
+  ret = rmw_validate_namespace(namespace_, &validation_result, nullptr);
+  if (RMW_RET_OK != ret || RMW_NAMESPACE_VALID != validation_result) {
+    const char * reason = RMW_RET_OK == ret ?
+      rmw_node_name_validation_result_string(validation_result) :
+      rmw_get_error_string().str;
+    RMW_SET_ERROR_MSG_WITH_FORMAT_STRING("invalid node namespace: %s", reason);
     return nullptr;
   }
 
-  rmw_node_t * node_handle = nullptr;
   auto common_context = static_cast<rmw_dds_common::Context *>(context->impl->common);
   rmw_dds_common::GraphCache & graph_cache = common_context->graph_cache;
-
-  node_handle = rmw_node_allocate();
-  if (!node_handle) {
-    RMW_SET_ERROR_MSG("failed to allocate rmw_node_t");
-    goto fail;
+  rcutils_allocator_t allocator = context->options.allocator;
+  rmw_node_t * node = reinterpret_cast<rmw_node_t *>(
+    allocator.zero_allocate(1u, sizeof(rmw_node_t), allocator.state));
```

Comment:
:+1: for using the allocator in `context->options`, but is this change related?
what was the idea of the original `rmw_node_allocate()`?

Reply:
Yes and no. We don't have to, but I think we should settle on https://github.com/ros2/rmw/pull/249#issuecomment-655150826.

Reply:
I see what you mean about this not using the allocator properly.  But I think that is a separate topic, and so I would suggest leaving this as-is and we can discuss that separately.

Reply:
Fair enough.